docs/harvester-network.md

```diff
@@ -0,0 +1,70 @@
+
+### Implementation
+
+#### Management Network
+
+Harvester adopts [flannel](https://github.com/flannel-io/flannel) as the default CNI to implement the management network. It's an internal network, which means the user can only access the VM's management network within its cluster nodes or pods.
+
+#### VLAN
+
+[Harvester network-controller](https://github.com/rancher/harvester-network-controller) leverages the [multus](https://github.com/k8snetworkplumbingwg/multus-cni) and [bridge](https://www.cni.dev/plugins/current/main/bridge/) CNI plugins to implement the VLAN.  
+
+Below is a use case of the VLAN in Harvester.
+
+![image-20210412181210146](https://raw.githubusercontent.com/yaocw2020/img/master/image-20210412181210146.png)
+
+- Harvester network controller use a bridge for a node and a pair of veth for a VM to implement the VLAN. The bridge acts as a switch to forward the network traffic from or to VMs and the veth pair is like the connected ports between vms and switch.
+- VMs within the same VLAN is able to communicate with each other, while the VMs within different VLANs can't.
+- The external switch ports connected with the hosts or other devices(such as DHCP server) should be set as trunk or hybrid type and permit the specified VLANs.
+
+### UI Interaction
+
+- Enable VLAN via going to **Setting > vlan** to enable VLAN and input a valid default physical NIC name for the VLAN. The first physical NIC name of each Harvester node always defaults to eth0. It is recommended to choose a separate NIC for the VLAN other than the one used for the management network(the one selected during the Harvester installation) for better network performance and isolation. (Note: modify the default VLAN network setting will not change the existing configured host networks).
+
+  ![image-20210408233858097](https://raw.githubusercontent.com/yaocw2020/img/master/image-20210408233858097.png)
+
+- (optional) Users can always customize each node's VLAN network configuration via going to the **HOST > Network** tab.
+
+  ![image-20210408235822223](https://raw.githubusercontent.com/yaocw2020/img/master/image-20210408235822223.png)
+
+- Creating a new VLAN network via going to the **Advanced>Networks** page and clicking the **Create** button.
+
+  ![image-20210408232931111](https://raw.githubusercontent.com/yaocw2020/img/master/image-20210408232931111-20210409000151223.png)
+
+- Creating a VM and add the network configurations.
+
+  - Only the first network card will be enabled by default, the user can either choose to use a management network or VLAN network. (Note: will need to select `Install guest agent` option in the `Advanced Options` tab to get the VLAN network IP address from the Harvester UI)
+
+    ![image-20210412175826097](https://raw.githubusercontent.com/yaocw2020/img/master/image-20210412175826097.png)
+
+  - Users can choose to add one or multiple network cards, the additional network card configurations can be set via cloud-init network data. e.g.
```

Comment:
Users can choose to add one or multiple network interface cards. Additional network interface card configurations can be set via cloud-init network data.